IDE/VisualStudio/README.md

```diff
@@ -0,0 +1,266 @@
+
+
+### Studio Command Prompt
+
+Select `View` - `Terminal` from the menu bar.
+
+* Configure: `cmake --preset <preset name>`
+* Build: `cmake --build --preset <preset name>`
+
+```bash
+# delete build directory
+rmdir /s /q build-stm32l4
+
+# configure
+cmake --preset stm32l4
+
+# build
+cmake --build --preset stm32l4
+```
+
+If there are no devices listed in the `Manage Configurations` drop-down, ensure the `CMakePresets.json` is valid.
+A single json syntax error will spoil the entire project.
+
+## Local Visual Studio Projects
+
+Project files can be found in [`[WOLFBOOT_ROOT]/IDE/VisualStudio`](../../IDE/VisualStudio/README.md).
+
+There are three projects to:
+
+1. Generate a new signing key
+2. Sign an image
+3. Verify the signed image
+
+Visual Studio `$(ProjectDir)` is typically `[WOLFBOOT_ROOT]\tools\keytools`.
+
+-----
+
+## Keytools Usage
+
+See [keytools/README.md](../../keytools/README.md)
+
+### Step 1: wolfBootKeyGenTool Visual Studio Project
+
+Build the project. Generate a new signing key with `keygen.exe`.
+
+```DOS
+keygen.exe [ --ed25519 | --ed448 | --ecc256 | --ecc384 | --ecc521 | --rsa2048 | --rsa3072 | --rsa4096 ] ^
+           [ -g privkey]     [ -i pubkey] [ -keystoreDir dir]  ^
+           [ --id {list}]    [ --der]                        ^
+           [ --exportpubkey] [ --nolocalkeys]
+```
+
+WARNING: Key Generation will *overwrite* any prior keystore files.
+
+Right-click on `wolfBootKeygenTool` project, typically in:
+
+```text
+C:\workspace\wolfBoot-%USERNAME%\tools\keytools
+```
+
+Select: Properties - Configuration Properties - Debugging:
+
+```text
+Command:           $(TargetPath)
+Command Arguments: --ed25519 -g $(ProjectDir)wolfboot_signing_private_key.der   -keystoreDir  $(ProjectDir)
+Working Directory: $(ProjectDir)..\..\
+```
+
+Replace `$(ProjectDir)` with your desired `keystoreDir` path for keys and firmware locations.
+Otherwise the private key will be created in the project directory `[WOLFBOOT_ROOT]\tools\keytools`.
+
+Example:
+
+```DOS
+cd %WOLFBOOT_ROOT%\tools\keytools
+
+:: cmd       sign     private key
+:: ------- --------- -----------------------------------
+keygen.exe --ed25519 -g wolfboot_signing_private_key.der
+```
+
+Expected output:
+
+```text
+wolfBoot KeyGen
+Keystore size:        2608
+Saving keystore file: C:\workspace\wolfBoot-gojimmypi\tools\keytools\/keystore.c
+Selected Keytype:     ECC256
+Generating key (type: ECC256)
+Associated key file:  C:\workspace\wolfBoot-gojimmypi\tools\keytools\wolfboot_signing_private_key.der
+Partition ids mask:   ffffffff
+Key type:             ECC256
+Public key slot:      0
+Done.
+```
+
+-----
+
+### Step 2: wolfBootSignTool Visual Studio Project
+
+Build the project. Sign an image with `sign.exe  [OPTIONS]  IMAGE.BIN  KEY.DER  VERSION`.
+
+Right-click on `wolfBootSignTool` project, typically in:
+
+```text
+C:\workspace\wolfBoot-%USERNAME%\tools\keytools
```

Comment:
Seems odd to include the full path example. Let users assume they know where the cloned/downloaded wolfboot is located.